File: src/karabo/util/Trainstamp.cc
/* 
 * File:   TrainStamp.cc
 * Author: WP76
 * 
 * Created on June 19, 2013, 3:22 PM
 */

#include "Trainstamp.hh"

namespace karabo {
    namespace util {


        Trainstamp::Trainstamp() : m_trainId(0) {
        }


        Trainstamp::Trainstamp(const unsigned long long trainId) : m_trainId(trainId) {
        }


        Trainstamp::~Trainstamp() {
        }


        bool Trainstamp::hashAttributesContainTimeInformation(const Hash::Attributes attributes) {
            return attributes.has("tid");
        }


        Trainstamp Trainstamp::fromHashAttributes(const Hash::Attributes& attributes) {
            unsigned long long tid;
            try {
                attributes.get("tid", tid);
            } catch (const Exception& e) {
                KARABO_RETHROW_AS(KARABO_PARAMETER_EXCEPTION("Provided attributes do not contain proper trainId information"));
            }
            return Trainstamp(tid);
        }


        void Trainstamp::toHashAttributes(Hash::Attributes& attributes) const {
            attributes.set("tid", m_trainId);
        }
    }
}

File: src/jkarabo/karabo/util/time/Trainstamp.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package karabo.util.time;

import karabo.util.Attributes;

/**
 *
 * @author Sergey Esenov serguei.essenov at xfel.eu
 */
public class Trainstamp {

    private long m_trainId;

    public Trainstamp() {
    }

    public Trainstamp(long trainId) {
        m_trainId = trainId;
    }

    public long getTrainId() {
        return m_trainId;
    }
    
    public static boolean hashAttributesContainTimeInformation(Attributes attributes) {
        return attributes.has("tid");
    }
    
    public static Trainstamp fromHashAttributes(Attributes attributes) {
        long tid;
        try {
            tid = (long) attributes.get("tid");
        } catch(Exception e) {
            throw new RuntimeException("Failed to access \"tid\" attribute: " + e);
        }
        return new Trainstamp(tid);
    }
    
    public void toHashAttributes(Attributes attributes) {
        attributes.set("tid", m_trainId);
    }
}
